src/en/credentials.md

```diff
@@ -214,12 +226,90 @@ juju add-credential aws -f mycreds.yaml --replace
 This will overwrite existing credential information, so make sure all current
 credentials are contained in the file, not just the new or changed one.
 
+Replacing credentials in this way does not update credentials currently in use
+(on an existing controller/cloud). See next section for that. The
+`add-credential` command is always "pre-bootstrap" in nature.
+
+### Updating credentials
+
+To update credentials currently in use (i.e. cached on the controller) the
+`update-credential` command is used. The requirements for using this command,
+as compared to the initial `juju bootstrap` (or `juju add-model`) command, are:
+
+ - same cloud name
+ - same Juju username (logged in)
+ - same credential name
+
+The update is a two-step process. First change the credentials locally with the
+`add-credential` command (in conjunction with the `--replace` option) and then
+upload those credentials to the controller.
+
+Below, we log in with the correct Juju username (e.g. 'admin'), change the
+contents of the credential called 'joe', and then update them on a Google cloud
+controller:
+
+```bash
+juju add-credential --replace joe
+juju update-credential google joe
+```
+
+!!! Warning:
+    It is not possible to update the credentials if the initial credential name
+    is unknown. This restriction will be removed in an upcoming release of
+    Juju.
+
+####  Updating credentials with a different Juju user
+
+If you are unable to ascertain the original Juju username then you will need
+to use a different one. This implies adding a new credential name, copying over
+any authentication material into the old credential name, and finally updating
+the credentials. Below we demonstrate this for the Azure cloud:
+
+Add a new temporary credential name (like 'new-credential-name') and gather all
+credential sets (new and old):
+
+```bash
+juju add-credential azure
+juju credentials azure --format yaml --show-secrets > azure-creds.yaml
+```
+
+Copy the values of `application-id` and `application-password` from the new set
+to the old set.
+
+Then replace the local credentials and upload them to the controller:
+
+```bash
+juju add-credential azure -f azure-creds.yaml --replace
+juju update-credential azure old-credential-name
+```
+
+To be clear, the file `azure-creds.yaml` (used with `add-credential`) should
+look similar to:
+
+```no-highlight
+Credentials:
+  azure:
+    new-credential-name:
+      auth-type: service-principal-secret
+      application-id: foo1
+      application-password: foo2
+      subscription-id: bar
+    old-credential-name:
+      auth-type: service-principal-secret
+      application-id: foo1
+      application-password: foo2
+      subscription-id: bar
+```
+
+### Removing credentials
+
 If a credential is no longer required, it can be removed:
```

Comment:
This should clarify - 'locally'. 'remove-credential' command only operates on a client- not on the controller-stored credentials.

Reply:
:ok: